From david-b@pacbell.net Wed Nov  1 14:26:46 2006
From: David Brownell <david-b@pacbell.net>
To: linux-omap-open-source@linux.omap.com
Subject: USB: fix compiler issues with newer gcc versions
Date: Wed, 1 Nov 2006 14:26:26 -0800
Cc: Andrew Morton <akpm@osdl.org>, "Arnaud Patard (Rtp)" <arnaud.patard@rtp-net.org>, Felipe Balbi <ext-felipe.lima@nokia.com>, Greg KH <greg@kroah.com>, linux-usb-devel@lists.sourceforge.net
Message-Id: <200611011426.28869.david-b@pacbell.net>

Remove complaint from newer GCCs; they don't like forward function
declarations except in top-level contexts.

Signed-off-by: David Brownell <dbrownell@users.sourceforge.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/core/hub.c |    3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

--- gregkh-2.6.orig/drivers/usb/core/hub.c
+++ gregkh-2.6/drivers/usb/core/hub.c
@@ -1188,6 +1188,7 @@ static inline void show_string(struct us
 
 #ifdef	CONFIG_USB_OTG
 #include "otg_whitelist.h"
+static int __usb_port_suspend(struct usb_device *, int port1);
 #endif
 
 /**
@@ -1289,8 +1290,6 @@ int usb_new_device(struct usb_device *ud
 		 * (Includes HNP test device.)
 		 */
 		if (udev->bus->b_hnp_enable || udev->bus->is_b_host) {
-			static int __usb_port_suspend(struct usb_device *,
-						int port1);
 			err = __usb_port_suspend(udev, udev->bus->otg_port);
 			if (err < 0)
 				dev_dbg(&udev->dev, "HNP fail, %d\n", err);
